### PR TITLE
Refuse to delete a repository in OpenProject on FS permission errors

### PR DESCRIPTION
**Pull request by oliverguenther**
This commit changes the behavior of removing repositories.
It is no longer possible to remove a managed repository (model) when the
underlying repository directory could not be removed.

This error originally occurred when Apache committed changes to the repository
through mod_svn, which resulted in the OpenProject user being unable to
delete a particular subdirectory.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-repository-deletion.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-repository-deletion.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/controllers/repositories_controller.rb [hunk 1]
```diff
@@ -109,8 +109,12 @@ def destroy_info
   end
 
   def destroy
-    @project.repository.destroy
-    flash[:notice] = I18n.t('repositories.delete_sucessful')
+    repository = @project.repository
+    if repository.destroy
+      flash[:notice] = I18n.t('repositories.delete_sucessful')
+    else
+      flash[:error] = repository.errors.full_messages.join("\n")
```
Comment (NobodysNightmare) [inline app/controllers/repositories_controller.rb]: I could not believe that `flash` was clever enough to turn `\n` into actal HTML line breaks, so I tried it out:

``` ruby
flash[:notice] = ['a', 'b', 'c'].join("\n")
```

Sadly the result was as I expected:
image

**tl; dr:** You should either use `<br>` or some other clever trick... I believe in error flashs we are usually using unordered lists...

File: app/services/scm/delete_managed_repository_service.rb [hunk 2]
```diff
@@ -60,8 +60,8 @@ def delete_repository
 
     true
   rescue SystemCallError => e
-    Rails.logger.error("An error occurred while accessing the repository '#{repository.root_url}' \
-                        on filesystem: #{e.message}")
+    Rails.logger.error("An error occurred while accessing the repository '#{repository.root_url}'" +
+                       "on filesystem: #{e.message}")
```
Comment (NobodysNightmare) [inline app/services/scm/delete_managed_repository_service.rb]: missing space between `'` and `on`.

File: config/locales/en.yml [hunk 3]
```diff
@@ -1317,7 +1317,8 @@ en:
       build_failed: "Unable to create the repository with the selected configuration. %{reason}"
       empty_repository: "The repository exists, but is empty. It does not contain any revisions yet."
       exists_on_filesystem: "The repository directory exists already on filesystem."
-      filesystem_access_failed: "An error occurred while accessing the repository on filesystem: %{message}."
+      filesystem_access_failed: "An error occurred while accessing the repository on filesystem: %{message}"
+      deletion_failed: "The repository could not be deleted on disk due to lacking permissions. Please contact your administrator to resolve this issue."
```
Comment (NobodysNightmare) [inline config/locales/en.yml]: Correct me if I am wrong, because I sometimes get those pronouns wrong, but shouldn't it be _deleted_ **_from**_ _disk_?

File: app/controllers/repositories_controller.rb (hunk 1 above)
Comment (oliverguenther) [inline app/controllers/repositories_controller.rb]: Strange, as this is used quite a number of controllers (search for `errors.full_messages.join("\n")`). 

What does work is `errors.full_messages`,  as the flash render helper joins them into `<br/>`. (cf. `join_flash_messages`)

File: config/locales/en.yml (hunk 3 above)
Comment (oliverguenther) [inline config/locales/en.yml]: I think both technically work, but I find more results for _delete from x_. Will change that.

**Comment by oliverguenther** [comment]
@NobodysNightmare Thanks for your remarks. I've addressed them in the latest commit. Feel free to merge when travis agrees.

**Comment by NobodysNightmare** [comment]
LGTM :+1: 

Waiting for travis.
